File: Source/Fantasy.Jobs/Scheduling/TimeTrigger.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Serialization;
using Fantasy.XSerialization;

namespace Fantasy.Jobs.Scheduling
{
    [DataContract(Name="Time")]
    [XSerializable("timeTrigger", NamespaceUri = Consts.ScheduleNamespaceURI)]
    public class TimeTrigger : Trigger 
    {

        public TimeTrigger()
        {
            this.Date = DateTime.Now.Date;
        }

        [DataMember] 
        [XAttribute("date")]
        public DateTime Date { get; set; }

        public override TriggerType Type
        {
            get
            {
                return TriggerType.Time;
            }
        }



        protected override string TriggerTimeDescription
        {
            get 
            {
                DateTime t = this.Date + this.StartTime;
                return string.Format("At {0:t} on {0:d}.", t); 
            }
        }
    }
}

File: branches/Stheno/Fantasy.Jobs/src/fantasy/jobs/scheduling/TimeTrigger.java
package fantasy.jobs.scheduling;

import java.util.Calendar;

import org.apache.commons.lang3.time.DateUtils;
import org.apache.commons.lang3.time.FastDateFormat;

import fantasy.TimeSpan;
import fantasy.xserialization.*;

@XSerializable(name = "timeTrigger", namespaceUri = fantasy.jobs.Consts.ScheduleNamespaceURI)
public class TimeTrigger extends Trigger
{

	/**
	 * 
	 */
	private static final long serialVersionUID = -6686895968828707760L;
	public TimeTrigger()
	{
		this.setDate(DateUtils.truncate(new java.util.Date(), Calendar.DATE));
	}

    @XAttribute(name = "date")
	private java.util.Date privateDate;
	public final java.util.Date getDate()
	{
		return privateDate;
	}
	public final void setDate(java.util.Date value)
	{
		privateDate = value;
	}

	@Override
	public TriggerType getType()
	{
		return TriggerType.Time;
	}



	@Override
	protected String getTriggerTimeDescription()
	{
		java.util.Date t = TimeSpan.add(this.getDate(), this.getStartTime());
		
		FastDateFormat tf = FastDateFormat.getTimeInstance(FastDateFormat.SHORT);
		FastDateFormat df = FastDateFormat.getDateInstance(FastDateFormat.SHORT);
		return String.format("At %1$s on %2$s.", tf.format(t), df.format(t));
	}
}